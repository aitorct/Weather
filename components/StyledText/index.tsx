/* eslint-disable react-native/no-unused-styles */

import React from "react";
import { StyleSheet, Text } from "react-native";
import colors from "@theme/colors";
import fonts from "@theme/fonts";

type TextStyle = "header" | "title" | "subtitle" | "button" | "text";

type Props = {
  style: TextStyle;
  children: string;
};

const StyledText = ({ style, children }: Props) => {
  const textStyle = [styles.base, styles[style]];

  return (
    <Text testID={`Text: ${children}`} style={textStyle}>
      {children}
    </Text>
  );
};

const styles = StyleSheet.create({
  header: fonts.header,
  title: fonts.title,
  subtitle: fonts.subtitle,
  text: fonts.text,
  button: {
    ...fonts.button,
    color: colors.buttonText,
  },
  base: {
    color: colors.text,
  },
});

export default StyledText;
