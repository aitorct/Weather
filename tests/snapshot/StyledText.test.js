import React from "react";
import renderer from "react-test-renderer";
import StyledText from "@components/StyledText";

describe("StyledText", () => {
  it("renders header style correctly", () => {
    const tree = renderer
      .create(<StyledText style="header">StyledText test</StyledText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders title style correctly", () => {
    const tree = renderer
      .create(<StyledText style="title">StyledText test</StyledText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders subtitle style correctly", () => {
    const tree = renderer
      .create(<StyledText style="subtitle">StyledText test</StyledText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders button style correctly", () => {
    const tree = renderer
      .create(<StyledText style="button">StyledText test</StyledText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders text style correctly", () => {
    const tree = renderer
      .create(<StyledText style="text">StyledText test</StyledText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
