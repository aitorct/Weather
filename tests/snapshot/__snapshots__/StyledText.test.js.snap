// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText renders button style correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#16133A",
      },
      {
        "color": "#E0E0E0",
        "fontSize": 24,
        "fontWeight": "500",
      },
    ]
  }
  testID="Text: StyledText test"
>
  StyledText test
</Text>
`;

exports[`StyledText renders header style correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#16133A",
      },
      {
        "fontSize": 48,
        "fontWeight": "bold",
      },
    ]
  }
  testID="Text: StyledText test"
>
  StyledText test
</Text>
`;

exports[`StyledText renders subtitle style correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#16133A",
      },
      {
        "fontSize": 24,
        "fontWeight": "500",
      },
    ]
  }
  testID="Text: StyledText test"
>
  StyledText test
</Text>
`;

exports[`StyledText renders text style correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#16133A",
      },
      {
        "fontSize": 18,
        "fontWeight": "bold",
      },
    ]
  }
  testID="Text: StyledText test"
>
  StyledText test
</Text>
`;

exports[`StyledText renders title style correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#16133A",
      },
      {
        "fontSize": 32,
        "fontWeight": "bold",
      },
    ]
  }
  testID="Text: StyledText test"
>
  StyledText test
</Text>
`;
